### PR TITLE
cmake/modules/Findpmem: always set pmem_VERSION_STRING

### PR DESCRIPTION
**Pull request by tchaikov**
before this change, `pmem_VERSION_STRING` is not set if it is not able
to fulfill the specified version requirement. the intention was to check
if the version is able to satisfy the requirement. but actually, passing
an empty `pmem_VERSION_STRING` to `find_package_handle_standard_args()`
as the option of `VERSION_VAR` does not fail this check. on the
contrary, it prints

-- Found pmem: pmem_pmemobj_INCLUDE_DIR;pmem_pmem_INCLUDE_DIR (Required
is at least version "1.17")

if we requires pmem 1.17, while the found version is, for instance,
1.10.

if the required version is 1.7, and the found version is 1.10, the
output from cmake is:

-- Found pmem: pmem_pmemobj_INCLUDE_DIR;pmem_pmem_INCLUDE_DIR (found
suitable version "1.10", minimum required is "1.7")

in this change,

* the version spec is not specified when calling `pkg_check_modules()`.
* we don't check the returned `PKG_${component}_VERSION` anymore,

so we can always delegate the version checking to
`find_package_handle_standard_args()`. please note, we use the lower
version returned by pkg-config if multiple components are required and
both pkg-config settings return their versions.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @CongMinYin

File: cmake/modules/Findpmem.cmake [hunk 1]
```diff
@@ -16,17 +16,10 @@ foreach(component ${pmem_FIND_COMPONENTS})
...
     message(FATAL_ERROR "unknown libpmem component: ${component}")
   endif()
-  set(pkg_module_spec "lib${component}")
-  if(pmem_FIND_VERSION_EXACT)
-    set(pkg_module_spec "lib${component}=${pmem_FIND_VERSION}")
-  elseif(pmem_FIND_VERSION)
-    set(pkg_module_spec "lib${component}>=${pmem_FIND_VERSION}")
-  endif()
-  pkg_check_modules(PKG_${component} QUIET ${pkg_module_spec})
+  pkg_check_modules(PKG_${component} QUIET "lib${component}")
```
Comment (cyx1231st) [inline cmake/modules/Findpmem.cmake:19]: Per my understanding, now `pmem_VERSION_STRING` will always be set to the minimun version of the found components. So the version check in `find_package_handle_standard_args()` can fail correctly if `pmem_VERSION_STRING` doesn't satisfy the user-specified version.

**Review by cyx1231st — APPROVED** [review]

File: ceph.spec.in [hunk 2]
```diff
@@ -29,12 +29,12 @@
 %else
 %bcond_without tcmalloc
 %endif
+%bcond_with system_pmdk
```
Comment (cyx1231st) [inline ceph.spec.in:32]: ack, `%bcond_with` means "disable by default".

**Comment by CongMinYin** [comment]
I'll test it tomorrow morning.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-cmake-findpmem-kefu/ad85af2ef8b76481adc0d1500033cd5094b8f21e/

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**